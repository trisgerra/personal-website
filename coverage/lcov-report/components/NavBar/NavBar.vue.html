<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/NavBar/NavBar.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/NavBar</a> NavBar.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;nav class="navbar" role="navigation" aria-label="main navigation"&gt;
    &lt;navigation-brand :switch-active="this.switchActive"&gt;&lt;/navigation-brand&gt;
    &lt;div :class="this.navBarMenuClass"&gt;
      &lt;div class="navbar-start"&gt;
        &lt;!-- navbar items --&gt;
        &lt;router-link
          v-for="route in routes"
          v-bind:key="route.name"
          class="menu-item"
          :to="route.path"
        &gt;
          &lt;a class="navbar-item navbar-router"&gt;{{route.name}}&lt;/a&gt;
        &lt;/router-link&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  &lt;/nav&gt;
&lt;/template&gt;
&nbsp;
&lt;script lang="ts"&gt;
import { Component, Prop, Vue } from "vue-property-decorator";
import { IRoute, routes } from "../../router";
import NavigationBrand from "./NavigationBrand.vue";
&nbsp;
@Component({
  components: {
    NavigationBrand
  }
})
export default class NavBar extends Vue {
  routes: IRoute[] = routes;
  isActive: string = "";
&nbsp;
  get navBarMenuClass(): string {
    return `navbar-menu ${this.isActive}`;
  }
  
<span class="fstat-no" title="function not covered" >  switchActive(): void {</span>
<span class="cstat-no" title="statement not covered" >    this.isActive = this.isActive === "is-active" ? "" : "is-active";</span>
  }
}
&lt;/script&gt;
&nbsp;
&lt;style scoped lang="scss"&gt;
.navbar {
  border-bottom: 1px solid #151616;
}
&nbsp;
.navbar-logo {
  height: 100%;
  background-color: #151616;
  border-left: 3px solid #151616;
  &amp;:hover {
    background-color: #151616;
    border-left: 3px solid #ff0000;
  }
}
&nbsp;
.navbar-router {
  background-color: white;
  font-weight: 500;
  border-bottom: 3px solid white;
  height: 100%;
&nbsp;
  &amp;:hover {
    color: #151616;
    font-weight: 600;
    border-bottom: 3px solid #151616;
  }
}
&lt;/style&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jul 07 2019 12:22:24 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
