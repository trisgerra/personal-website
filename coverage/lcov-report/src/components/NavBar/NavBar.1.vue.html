<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/NavBar/NavBar.1.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/NavBar</a> NavBar.1.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/7</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;nav class="navbar" role="navigation" aria-label="main navigation"&gt;
    &lt;div class="navbar-brand"&gt;
      &lt;router-link class="menu-item" to="/"&gt;
        &lt;div class="navbar-item navbar-logo"&gt;
          &lt;img src="../../assets/logo.png" alt="ADG Official" height="38" /&gt;
        &lt;/div&gt;
      &lt;/router-link&gt;
      &lt;a role="button" class="navbar-burger" aria-label="menu" aria-expanded="false" @click="this.switchActive"&gt;
        &lt;span aria-hidden="true"&gt;&lt;/span&gt;
        &lt;span aria-hidden="true"&gt;&lt;/span&gt;
        &lt;span aria-hidden="true"&gt;&lt;/span&gt;
      &lt;/a&gt;
    &lt;/div&gt;
    &lt;div :class="this.navBarMenuClass"&gt;
      &lt;div class="navbar-start"&gt;
        &lt;!-- navbar items --&gt;
        &lt;router-link
          v-for="route in routes"
          v-bind:key="route.name"
          class="menu-item"
          :to="route.path"
        &gt;
          &lt;a class="navbar-item navbar-router"&gt;{{route.name}}&lt;/a&gt;
        &lt;/router-link&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  &lt;/nav&gt;
&lt;/template&gt;
&nbsp;
&lt;script lang="ts"&gt;
<span class="cstat-no" title="statement not covered" >import { Component, Prop, Vue } from "vue-property-decorator";</span>
<span class="cstat-no" title="statement not covered" >import { IRoute, routes } from "../../router";</span>
&nbsp;
<span class="fstat-no" title="function not covered" >@Component</span>
<span class="cstat-no" title="statement not covered" >export default class NavBar extends Vue {</span>
<span class="cstat-no" title="statement not covered" >  routes: IRoute[] = routes;</span>
<span class="cstat-no" title="statement not covered" >  isActive: string = '';</span>
&nbsp;
<span class="fstat-no" title="function not covered" >  get navBarMenuClass() : string {</span>
<span class="cstat-no" title="statement not covered" >    return `navbar-menu ${this.isActive}`</span>
  }
<span class="fstat-no" title="function not covered" >  switchActive(): void {</span>
<span class="cstat-no" title="statement not covered" >    this.isActive = this.isActive === 'is-active' ? '' : 'is-active';</span>
  }
  
}
&lt;/script&gt;
&nbsp;
&lt;style scoped lang="scss"&gt;
.navbar {
  border-bottom: 1px solid #151616;
}
&nbsp;
.navbar-logo {
  height: 100%;
  background-color: #151616;
  border-left: 3px solid #151616;
  &amp;:hover {
    background-color: #151616;
    border-left: 3px solid #ff0000;
  }
}
&nbsp;
.navbar-router {
  background-color: white;
  font-weight: 500;
  border-bottom: 3px solid white;
  height: 100%;
&nbsp;
  &amp;:hover {
    color: #151616;
    font-weight: 600;
    border-bottom: 3px solid #151616;
  }
}
&lt;/style&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jul 07 2019 11:17:08 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
