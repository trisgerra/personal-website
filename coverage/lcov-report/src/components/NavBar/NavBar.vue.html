<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/NavBar/NavBar.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/NavBar</a> NavBar.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;nav class="navbar" role="navigation" aria-label="main navigation"&gt;
    &lt;div class="navbar-brand"&gt;
      &lt;router-link class="menu-item" to="/"&gt;
        &lt;img class="logo" src="../../assets/logo.png" alt="ADG Official" height="38" /&gt;
      &lt;/router-link&gt;
      &lt;a role="button" class="navbar-burger" aria-label="menu" aria-expanded="false"&gt;
        &lt;span aria-hidden="true"&gt;&lt;/span&gt;
        &lt;span aria-hidden="true"&gt;&lt;/span&gt;
        &lt;span aria-hidden="true"&gt;&lt;/span&gt;
      &lt;/a&gt;
    &lt;/div&gt;
    &lt;div class="navbar-menu"&gt;
      &lt;div class="navbar-start"&gt;
        &lt;!-- navbar items --&gt;
        &lt;a class="navbar-item"&gt;Blog&lt;/a&gt;
        &lt;a class="navbar-item"&gt;Design&lt;/a&gt;
        &lt;a class="navbar-item"&gt;Projects&lt;/a&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  &lt;/nav&gt;
&lt;/template&gt;
&nbsp;
&lt;script lang="ts"&gt;
import { Component, Prop, Vue } from 'vue-property-decorator';
&nbsp;
@Component
export default class NavBar extends Vue {}
&lt;/script&gt;
&nbsp;
&lt;style scoped lang="scss"&gt;
.router-link-active, .router-link-exact-active, .navbar-brand {
  width: fit-content;
}
.navbar {
  height: 50px;
  border-bottom: 1px solid #151616;
}
.logo {
  height: 100%;
  padding: 12px;
  background-color: #151616;
  &amp;:hover {
    background-color: #151616;
    border-left: 3px solid #ff0000;
  }
}
.navbar-item {
  background-color: white;
  font-weight: 400;
  &amp;:hover {
    color: #151616;
    font-weight: 600;
    border-bottom: 3px solid #151616;
  }
}
&lt;/style&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jul 07 2019 00:51:08 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
