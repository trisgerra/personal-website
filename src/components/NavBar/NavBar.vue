<template>
  <nav class="navbar" role="navigation" aria-label="main navigation">
    <div class="navbar-brand">
      <router-link class="menu-item" to="/">
        <img class="logo" src="../../assets/logo.png" alt="ADG Official" height="38" />
      </router-link>
      <a role="button" class="navbar-burger" aria-label="menu" aria-expanded="false">
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
      </a>
    </div>
    <div class="navbar-menu">
      <div class="navbar-start">
        <!-- navbar items -->
        <a class="navbar-item">Blog</a>
        <a class="navbar-item">Design</a>
        <a class="navbar-item">Projects</a>
      </div>
    </div>
  </nav>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class NavBar extends Vue {}
</script>

<style scoped lang="scss">
.router-link-active, .router-link-exact-active, .navbar-brand {
  width: fit-content;
}
.navbar {
  height: 50px;
  border-bottom: 1px solid #151616;
}
.logo {
  height: 100%;
  padding: 12px;
  background-color: #151616;
  &:hover {
    background-color: #151616;
    border-left: 3px solid #ff0000;
  }
}
.navbar-item {
  background-color: white;
  font-weight: 400;
  &:hover {
    color: #151616;
    font-weight: 600;
    border-bottom: 3px solid #151616;
  }
}
</style>
