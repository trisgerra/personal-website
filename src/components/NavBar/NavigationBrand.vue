<template>
  <div class="navbar-brand">
    <router-link class="menu-item" to="/">
      <div class="navbar-item navbar-logo">
        <img src="../../assets/logo.png" alt="ADG Official" height="38" />
      </div>
    </router-link>
    <a
      role="button"
      class="navbar-burger"
      aria-label="menu"
      aria-expanded="false"
      @click="this.switchActive">
      <span aria-hidden="true"></span>
      <span aria-hidden="true"></span>
      <span aria-hidden="true"></span>
    </a>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { IRoute, routes } from '../../router';

@Component
export default class NavigationBrand extends Vue {
  @Prop() switchActive!: Function;
}
</script>

<style scoped lang="scss">

.navbar-logo {
  height: 100%;
  background-color: #151616;
  border-left: 3px solid #151616;
  &:hover {
    background-color: #151616;
    border-left: 3px solid #ff0000;
  }
}
</style>
