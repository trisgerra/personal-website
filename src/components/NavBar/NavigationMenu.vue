<template>
  <div :class="this.navBarMenuClass">
    <div class="navbar-start">
      <router-link
        v-for="route in routes"
        v-bind:key="route.name"
        class="menu-item"
        :to="route.path">
        <a class="navbar-item navbar-router">{{route.name}}</a>
      </router-link>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { IRoute, routes } from '../../router';

@Component
export default class NavigationMenu extends Vue {
  @Prop() isActive!: Function;

  routes: IRoute[] = routes;

  get navBarMenuClass(): string {
    return this.isActive ? 'navbar-menu is-active' : 'navbar-menu';
  }
}
</script>

<style scoped lang="scss">
@import '~bulma/sass/utilities/mixins';

.navbar-router {
  background-color: white;
  font-weight: 400;
  height: 100%;
  color: #151616;
  &:hover {
    color: #151616;
    font-weight: 600;
  }
}

.router-link-exact-active {
  a {
    background-color: white;
    font-weight: 600;
  }
}
@include desktop {
  .navbar-router {
      border-bottom: 3px solid white;
      &:hover {
        border-bottom: 3px solid #151616;
      }
    }
    .router-link-exact-active {
        border-bottom: 3px solid #151616;
    }
}

@include touch {
  .navbar-router {
      border-left: 2px solid white;
      &:hover {
        border-left: 2px solid #151616;
      }
    }
}
</style>
