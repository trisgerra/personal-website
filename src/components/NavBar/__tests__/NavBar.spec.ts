import { Wrapper } from '@vue/test-utils';
import NavBar from '../NavBar.vue';
import createWrapper from '@/utils/CreateWrapper';

describe('NavBar', () => {
  it('should render as expected', () => {
    const wrapper: Wrapper<NavBar> = createWrapper(NavBar);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('contains a navigation bar', () => {
    const wrapper: Wrapper<NavBar> = createWrapper(NavBar);
    expect(wrapper.findAll('.navbar').length).toBe(1);
  });

  it('contains a logo', () => {
    const wrapper: Wrapper<NavBar> = createWrapper(NavBar);
    expect(wrapper.findAll('.logo').length).toBe(1);
  });

  it('contains an hamburger menu', () => {
    const wrapper: Wrapper<NavBar> = createWrapper(NavBar);
    expect(wrapper.findAll('.navbar-burger').length).toBe(1);
  });

  it('contains the menu items', () => {
    const wrapper: Wrapper<NavBar> = createWrapper(NavBar);
    expect(wrapper.findAll('.navbar-menu').length).toBe(1);
    expect(wrapper.findAll('.navbar-start').length).toBe(1);
  });
});
