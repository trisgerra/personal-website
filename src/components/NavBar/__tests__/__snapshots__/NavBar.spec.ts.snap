// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render as expected 1`] = `
<nav
  aria-label="main navigation"
  class="navbar"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <a
      class="menu-item router-link-exact-active router-link-active"
      href="#/"
    >
      <img
        alt="ADG Official"
        class="logo"
        height="38"
        src="../../assets/logo.png"
      />
    </a>
     
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger"
      role="button"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu"
  >
    <div
      class="navbar-start"
    >
      <a
        class="navbar-item"
      >
        Blog
      </a>
       
      <a
        class="navbar-item"
      >
        Design
      </a>
       
      <a
        class="navbar-item"
      >
        Projects
      </a>
    </div>
  </div>
</nav>
`;
